sandpack-react/src/presets/Sandpack.tsx

```diff
@@ -51,122 +64,198 @@ export const Sandpack: SandpackInternal = (props) => {
+    />
+  ) : undefined;
 
   /**
-   * Parts are set as `flex` values, so they set the flex shrink/grow
-   * Cannot use width percentages as it doesn't work with
-   * the automatic layout break when the component is under 700px
+   * Resizable
    */
-  const editorPart = props.options?.editorWidthPercentage || 50;
-  const previewPart = 100 - editorPart;
+  /* eslint-disable @typescript-eslint/no-explicit-any */
+  const dragEventTargetRef = React.useRef<any>(null);
 
-  const hasRightColumn =
-    props.options?.showConsole || props.options?.showConsoleButton;
-  const RightColumn = hasRightColumn ? SandpackStack : React.Fragment;
+  const [horizontalSize, setHorizontalSize] = React.useState(
+    options.editorWidthPercentage
+  );
+  const [verticalSize, setVerticalSize] = React.useState(70);
 
+  const RightColumn = hasRightColumn ? SandpackStack : React.Fragment;
   const rightColumnStyle = {
-    flexGrow: previewPart,
-    flexShrink: previewPart,
+    flexGrow: 100 - horizontalSize,
+    flexShrink: 100 - horizontalSize,
     flexBasis: 0,
-    minWidth: 700 * (previewPart / (previewPart + editorPart)),
     gap: consoleVisibility ? 1 : 0,
-    height: props.options?.editorHeight, // use the original editor height
+    height: options.editorHeight, // use the original editor height
   };
+  const topRowStyle = hasRightColumn
+    ? {
+        flexGrow: verticalSize,
+        flexShrink: verticalSize,
+        flexBasis: 0,
+        overflow: "hidden",
+      }
+    : rightColumnStyle;
 
-  const rightColumnProps = React.useMemo(
-    () => (hasRightColumn ? { style: rightColumnStyle } : {}),
-    [hasRightColumn]
-  );
+  const onDragMove = (event: MouseEvent): void => {
+    if (!dragEventTargetRef.current) return;
 
-  /* eslint-disable-next-line @typescript-eslint/no-non-null-assertion */
-  const templateFiles = SANDBOX_TEMPLATES[props.template!] ?? {};
-  const mode = "mode" in templateFiles ? templateFiles.mode : "preview";
+    const container = dragEventTargetRef.current
+      .parentElement as HTMLDivElement;
+    const direction = dragEventTargetRef.current.dataset.direction as
+      | "horizontal"
+      | "vertical";
+    const isHorizontal = direction === "horizontal";
 
-  const actionsChildren = props.options?.showConsoleButton ? (
-    <ConsoleCounterButton
-      counter={counter}
-      onClick={(): void => setConsoleVisibility((prev) => !prev)}
-    />
-  ) : undefined;
+    const { left, top, height, width } = container.getBoundingClientRect();
+    const offset = isHorizontal
+      ? ((event.clientX - left) / width) * 100
+      : ((event.clientY - top) / height) * 100;
+    const boundaries = Math.min(Math.max(offset, 25), 75);
+
+    if (isHorizontal) {
+      setHorizontalSize(boundaries);
+    } else {
+      setVerticalSize(boundaries);
+    }
+
+    container.querySelectorAll("iframe").forEach((frame) => {
+      frame.style.pointerEvents = "none";
```

Comment:
can't this be added directly on the preview component? to avoid the dom query here

Reply:
like sandpack state could know a dragging is in progress and propagate that to all components.

Reply:
In my opinion, the other components should be stateless and context-less as much as possible, plus, this resizable feature only works on Sandpack preset component in order to keep it isolated from the rest of the components.